cmd/gotestmd/gotestmd.go

```diff
@@ -60,22 +73,73 @@ func New() *cobra.Command {
 			}
 
 			suites := g.Generate(linkedExamples...)
-			for _, suite := range suites {
-				dir, _ := filepath.Split(suite.Location)
-				_ = os.MkdirAll(dir, os.ModePerm)
-				err := os.WriteFile(suite.Location, []byte(suite.String()), os.ModePerm)
-				if err != nil {
-					return errors.Errorf("cannot save suite %v, : %v", suite.Name(), err.Error())
-				}
+
+			if !bash {
+				return processGoSuites(suites)
 			}
 
-			return nil
+			matchRegex, err := regexp.Compile(match)
+			if err != nil {
+				return err
+			}
+
+			return processBashSuites(suites, matchRegex)
 		},
 	}
 
+	gotestmdCmd.Flags().Bool("bash", false, "generates bash scripts for tests. Can be used only with --match flag")
+	gotestmdCmd.Flags().String("match", "", "regex for matching suite or test name. Can be used only with --bash flag")
+
 	return gotestmdCmd
 }
 
+func processGoSuites(suites []*generator.Suite) error {
+	for _, suite := range suites {
+		dir, _ := filepath.Split(suite.Location)
+		_ = os.MkdirAll(dir, os.ModePerm)
+		err := os.WriteFile(suite.Location, []byte(suite.String()), os.ModePerm)
+		if err != nil {
+			return errors.Errorf("cannot save suite %v, : %v", suite.Name(), err.Error())
+		}
+	}
+
+	return nil
+}
+
+func processBashSuites(suites []*generator.Suite, matchRegex *regexp.Regexp) error {
+	for _, suite := range suites {
+		if !matchRegex.MatchString(suite.Name()) {
+			continue
+		}
+		suite.Tests = nil
+		dir, _ := filepath.Split(suite.Location)
+		_ = os.MkdirAll(dir, os.ModePerm)
+		err := os.WriteFile(suite.Location, []byte(suite.BashString()), os.ModePerm)
+		if err != nil {
+			return errors.Errorf("cannot save suite %v, : %v", suite.Name(), err.Error())
+		}
+		return nil
+	}
+
+	for _, suite := range suites {
+		for _, test := range suite.Tests {
+			if !matchRegex.MatchString(test.Name) {
+				continue
+			}
+			suite.Tests = []*generator.Test{test}
+			dir, _ := filepath.Split(suite.Location)
+			_ = os.MkdirAll(dir, os.ModePerm)
+			err := os.WriteFile(suite.Location, []byte(suite.BashString()), os.ModePerm)
+			if err != nil {
+				return errors.Errorf("cannot save suite %v, : %v", suite.Name(), err.Error())
+			}
+			return nil
+		}
+	}
+
+	return nil
```

Comment:
Currently if nothing matches, the program finishes without any error or warning.
I think we should print an error and return a non-zero status code when we didn't find the specified suite.

If several suites match the specified regexp, the program should either generate all of them, or return an error.